rpc/transaction.go

```diff
@@ -751,6 +752,8 @@ func adaptTransactionStatus(txStatus *starknet.TransactionStatus) (*TransactionS
 		status.Finality = TxnStatusAcceptedOnL2
 	case starknet.Received:
 		status.Finality = TxnStatusReceived
+	case starknet.NotReceived:
+		return nil, fmt.Errorf("%s", ErrTxnHashNotFound.Message)
```

Comment:
maybe it would better to create custom error and reuse it here and above? using rpc error (specifically error message) in my opinion a little bit tricky

Reply:
yep, sounds cleaner - updated